subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestWorker.java

```diff
@@ -123,6 +123,9 @@ public void stop() {
             processor.stop();
         } finally {
             completed.countDown();
+            // Clean the interrupted status
+            // because some test class processors do work here, e.g. JUnitPlatform
+            Thread.interrupted();
```

Comment:
⭕️ Why do we do this? What scenario would fail otherwise? Why hasn't this been a problem with testng, which has always done its work in the stop method?

Reply:
See [this test](https://github.com/gradle/gradle/blob/0ca79b7ca2ef5f72b57d068f694d5858c35eb127/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy#L41) and [the failure](https://builds.gradle.org/viewLog.html?buildId=10814445&tab=buildResultsDiv&buildTypeId=Gradle_Check_Quick_Java8_Oracle_Linux_testingJvm). The reason why testng doesn't fail is that this test is only testing JUnit 4 scenario.